modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js

```diff
@@ -0,0 +1,266 @@
+						gridField = createGridField();
+
+						gridField.setValue({});
+
+						assert.equal(gridField.getValue().A, undefined);
+
+						assert.equal(gridField.getValue().B, undefined);
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values if set an object with properties',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1', B: '2'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, '2');
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values if set an only option',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, undefined);
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values only if exists in the UI',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1', B: '2', C: '3'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, '2');
+
+						assert.equal(gridField.getValue().C, undefined);
+
+						assert.isObject(gridField.get('value'));
+
+						assert.equal(gridField.get('value').A, '1');
+
+						assert.equal(gridField.get('value').B, '2');
+
+						assert.equal(gridField.get('value').C, undefined);
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.getValue()',
+			function() {
+				it(
+					'should return object value if grid has a value in context',
+					function(done) {
+						gridField = createGridField(
+							{
+								value: {B: '2'}
+							}
+						);
+
+						var value = gridField.getValue();
+
+						assert.isObject(value);
+
+						assert.equal(value.B, '2');
+
+						done();
+					}
+				);
+
+				it(
+					'should return changed value object of the context with new object value',
+					function(done) {
+						gridField = createGridField(
+							{
+								value: {A: '1', B: '1'}
+							}
+						);
+
+						gridField.setValue({A: '2'});
+
+						var value = gridField.getValue();
+
+						assert.isObject(value);
+
+						assert.equal(value.A, '2');
+
+						assert.equal(value.B, undefined);
+
+						done();
+					}
+				);
+
+				it(
+					'should return the same context value if the grid began with value in context',
+					function(done) {
+						gridField = createGridField(
+							{
+								value: {A: '2', B: '1'}
+							}
+						);
+
+						var value = gridField.get('value');
+
+						assert.isObject(value);
+
+						assert.equal(value.A, '2');
+
+						assert.equal(value.B, '1');
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.checkItem()',
+			function() {
+				it(
+					'should click in a radio input and get its value',
```

Comment:
Add a test where a value is already set and you click in a different options to see if the value was changed